scheduler/server/stateful_scheduler.go

```diff
@@ -306,6 +311,11 @@ func NewStatefulScheduler(
 		return nil
 	}
 
+	dkef := durationKeyExtractorFn
```

Comment:
nit: why do we need this this assignment?

Reply:
this provides a default eliminating if fn == nil tests later in the code

Reply:
whoops didn't mean to resolve. i just meant - why can't we just say

`
if durationKeyExtractorFn == nil {
		durationKeyExtractorFn = nopDurationKeyExtractor
}
`

Reply:
I thought changing parameter values is a frowned upon practice.